### PR TITLE
Sp 7616 gw performance issue api changes proposal

### PR DESCRIPTION
**Pull request by przemyslawfierek-silvair**

### CODE REVIEW

File: doc/mesh-api.txt [hunk 1]
```diff
@@ -118,6 +118,40 @@ Methods:
...
+                uint16_t src;
+                uint16_t app_idx;
+                union {
+                    uint16_t net_idx;
+                    uint16_t dst;
+                    uint8_t  virt_label[16];
+                };
+                uint16_t size;
+                uint8_t  data[size];
+            };
```
Comment (michallowasrzechonek-silvair) [inline doc/mesh-api.txt:159]: This does not cover all possible combinations of `*MessageReceived` calls. You may have:
 - DevKeyMessageReceived(source=0x42, remote=false, net_index=1, data=...)
 - DevKeyMessageReceived(source=0x42, remote=true, net_index=1, data=...)
 - MessageReceived(source=0x42, key_index=1, destination=0x43, data=...)
 - MessageReceived(source=0x42, key_index=1, destination=<uuid>, data=...)

So the struct is more like:
```
struct {
    uint8_t element;
    uint8_t src;

    uint8_t type;
    union {
        struct {
            uint8_t net_idx;
            bool remote;
        } dev; // only meaningful when type is 0

        struct {
            uint8_t app_idx;
            uint16_t dst;
            uint8_t label[16]; // only meaningful when dst is virtual
        } app; // only meaningful when type is 1
    }

    uint8_t data[]; // note lacking 'size', you don't need it when using AF_UNIX, SOCK_DGRAM and recvmsg
};
```

File: doc/mesh-api.txt [hunk 2]
```diff
@@ -118,6 +118,40 @@ Methods:
 			org.bluez.mesh.Error.AlreadyExists,
 			org.bluez.mesh.Error.Failed
 
+    object node, array{byte, array{(uint16, dict)}} configuration, fd
+            Attach1(object app_root, uint64 token)
```
Comment (michallowasrzechonek-silvair) [inline doc/mesh-api.txt]: I don't like the name. We're not on windows ;P

File: doc/mesh-api.txt [hunk 3]
```diff
@@ -118,6 +118,40 @@ Methods:
...
+
+        This is an alternative version of the Attach() method. This method
+        acquires file descriptor for message reception
+        (which increases performance).
```
Comment (michallowasrzechonek-silvair) [inline doc/mesh-api.txt]: skip the comment about performance

File: doc/mesh-api.txt [hunk 4]
```diff
@@ -118,6 +118,40 @@ Methods:
...
+
+        Method behaviour and usage is similiar to the Attach() method.
+        The difference between those two method is that after call Attach1()
+        all received messages will be passed via file descriptor,
+        no more MessageReceived() and DevKeyMessageReceived() method
```
Comment (michallowasrzechonek-silvair) [inline doc/mesh-api.txt]: s/no more/instead of daemon calling ... methods/

File: doc/mesh-api.txt (hunk 2 above)
Comment (przemyslawfierek-silvair) [reply doc/mesh-api.txt]: Me either but I don't have better name. Do you think that `AcquireAttach` is better?

Comment (michallowasrzechonek-silvair) [reply doc/mesh-api.txt]: AttachFD, AttachStream, AttachSocket?

File: doc/mesh-api.txt [hunk 5]
```diff
@@ -118,6 +118,46 @@ Methods:
...
+
+        This is an alternative version of the Attach() method. This method
+        acquires file descriptor for message reception.
+
+        Method behaviour and usage is similiar to the Attach() method.
+        The difference between those two method is that after call AttachFD()
+        all received messages will be passed via file descriptor,
+        instead of daemon calling MessageReceived() and
+        DevKeyMessageReceived() methods.
+
```
Comment (michallowasrzechonek-silvair) [inline doc/mesh-api.txt:132]: Mention that the descriptor is a datagram socket - this is important from the user's pov, as the serialization protocol doesn't provide framing.

File: doc/mesh-api.txt [hunk 6]
```diff
@@ -118,6 +118,46 @@ Methods:
...
+                    struct {
+                        uint8_t app_idx;
+                        uint16_t dst;
+                        uint8_t label[16]; // only meaningful when dst is virtual
+                    } app; // only meaningful when type is 1
+                }
+
+                uint8_t data[];
+            };
```
Comment (michallowasrzechonek-silvair) [inline doc/mesh-api.txt:159]: Oh, and also mention endianness. I think since we're on the same machine, maybe use just host endianness, whatever it is?

**Pull request closed without merging**